### PR TITLE
add lesson05

### PR DESCRIPTION
**Pull request by zzsocool**
add except_exercise.py
add except_test.py
add mailroom_3.py

### CODE REVIEW

**Comment by natasha-aleksandrova** [comment]
@zzsocool github is showing merge conflict which will prevent me from merging -- can you merge with upstream please.

**Comment by natasha-aleksandrova** [comment]
if you need further help with github please let us know.

**Comment by zzsocool** [comment]
Sorry for the inconvenience. Conflict solved.

**Comment by zzsocool** [comment]
add mailroom_4.py
add test_mailroom.py

**Comment by natasha-aleksandrova** [comment]
thanks! will take another look

File: students/Z_shen/lesson06/test_mailroom.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    assert os.path.exists(file3)
+    assert os.path.exists(file4)
+
+
+if __name__ == '__main__':
+    test_donor_name_list()
+    test_add_new_donor()
+    test_add_amount_same_donor()
+    test_print_letter()
+    test_send_letter_to_all()
```
Comment (natasha-aleksandrova) [inline students/Z_shen/lesson06/test_mailroom.py:68]: make sure you use `pytest` command line utility to run your tests.

**Review by natasha-aleksandrova — APPROVED** [review]
nice work!

**Pull request merged**